### PR TITLE
Minimal buy button

### LINKED ISSUE #2: Minimal Buy Button
Parameters for 
* label
* price
* product name

Actions:
* trigger order process on shop #3

Assumptions for v1:
* Fixed style/CSS
* just one "global" shop object
* product name == product ID

Nice to have:
* optional product ID parameter

### PR DESCRIPTION
**Pull request by sectore**
incl.
- Styling options #9
- Polyfill for `CustomEvent`
- Clean up webpack config and other few things...

Fixes #2

### CODE REVIEW

File: src/components/checkout-button.ts [hunk 1]
```diff
@@ -1,127 +1,129 @@
...
+    switch (this.buttonColor) {
+      case 'light-blue':
+      case 'green':
+      case 'orange':
+      case 'red':
+      case 'gold':
+        clazz = this.buttonColor
+        break;
+      default: 
+        clazz = ''
```
Comment (svub) [inline src/components/checkout-button.ts]: I'm always a big fan of sharing a similar coding style.
I'd have gone for something (pseudo code)
```return `['green', 'orange', ...].contains(this.buttonColor) ? this.buttonColor: ''```
Wdyt? How do you feel about these choices?

File: src/components/checkout-button.ts [hunk 2]
```diff
@@ -1,127 +1,129 @@
...
+  static get styles() {
+    return [
+      themeStyles,
+      buttonStyles,
+      typographyStyles
+    ]
+  }
 
+  /**
+   * Name of the product to buy
```
Comment (svub) [inline src/components/checkout-button.ts]:
```suggestion
   * Optional label of the button - if not provided, product name will be used.
```

File: src/components/checkout-button.ts [hunk 3]
```diff
@@ -1,127 +1,129 @@
...
-  @property({type: String, attribute: 'button-class'}) 
-  buttonClass = 'nq-button';
+  name = '';
   
+  protected get hasName(): Boolean {
+    return this.name.trim().length > 0
+  }
+  
+  /**
+   * Price of the product to buy
```
Comment (svub) [inline src/components/checkout-button.ts]:
```suggestion
   * Price of the product to buy in NIM, minimum `0.00001`
```

File: src/components/shop.ts [hunk 4]
```diff
@@ -16,12 +16,23 @@ export class NimiqShop extends LitElement {
 
   connectedCallback() {
     super.connectedCallback()
-    // set address to checkout buttons
-    let buttons = this.getElementsByTagName('nimiq-shop-checkout-button')
+    const buttons = this.getElementsByTagName('nimiq-shop-checkout-button')
     for (let i = 0; i < buttons.length; i++) {
-      buttons[i].setAttribute('shop-address', this.address)
+      buttons[i].addEventListener('checkout', this.checkout);
```
Comment (svub) [inline src/components/shop.ts]:
```suggestion:-3
  for (const button of this.getElementsByTagName('nimiq-shop-checkout-button')) {
    button.addEventListener('checkout', this.checkout);
```
Again just, an idea. Please let me know what you think about suggestions like this.

Comment (svub) [reply src/components/shop.ts]: GitHub does not yet support multi-line suggestions. GitLab does. :-S
Move to GitLab? :D

File: src/index.html [hunk 5]
```diff
@@ -9,37 +9,58 @@
...
           </nimiq-shop-checkout-button>
         </div>
-    </div>
-</body>
+      </div>
+
+      <div class="nq-card">
+          <nimiq-shop-checkout-button
+            label="Small checkout button, disabled"
+            disable
```
Comment (svub) [inline src/index.html]: Not working for me, can still click it. But not crucial to have this feature for v1.

Comment (sectore) [reply src/index.html]: Which browser are you using? `Firefox` + `Chrome` are fine for me.

**Pull request merged**